--- nxt/usbsock.py.orig	2018-03-02 23:02:50.000000000 +0800
+++ nxt/usbsock.py	2018-03-02 23:04:22.000000000 +0800
@@ -19,6 +19,9 @@
 ID_VENDOR_LEGO = 0x0694
 ID_PRODUCT_NXT = 0x0002
 
+# Use longer timeout than default of 100 ms in PyUSB legacy.py
+TIMEOUT = 1000
+
 class USBSock(object):
     'Object for USB connection to NXT'
 
@@ -70,7 +73,7 @@
 
     def recv(self):
         'Use to recieve raw data over USB connection ***ADVANCED USERS ONLY***'
-        data = self.handle.bulkRead(self.blk_in.address, 64)
+        data = self.handle.bulkRead(self.blk_in.address, 64, TIMEOUT)
         if self.debug:
             print 'Recv:',
             print ':'.join('%02x' % (c & 0xFF) for c in data)
